packages/execution-environments/src/common/BaseSnapExecutor.ts

```diff
@@ -49,13 +55,39 @@ const fallbackError = {
   message: 'Execution Environment Error',
 };
 
-export type InvokeSnapArgs = SnapExportsParameters[0];
+// TODO: `KeyringParameters` expects a `chainId` for certain methods, but we're
+// not providing it in `getHandlerArguments`, resulting in type errors.
+export type InvokeSnapArgs = Omit<SnapExportsParameters[0], 'chainId'>;
 
 export type InvokeSnap = (
-  target: Target,
+  target: string,
   handler: HandlerType,
   args: InvokeSnapArgs | undefined,
-) => Promise<unknown>;
+) => Promise<Json>;
+
+/**
+ * The supported methods in the execution environment. The validator checks the
+ * incoming JSON-RPC request, and the `params` property is used for sorting the
+ * parameters, if they are an object.
+ */
+const EXECUTION_ENVIRONMENT_METHODS = {
+  ping: {
+    validator: PingRequestArgumentsStruct,
```

Comment:
```suggestion
    struct: PingRequestArgumentsStruct,
```

Would that make more sense?